### PR TITLE
Theta

### PR DESCRIPTION
**Pull request by gwenchee**

### CODE REVIEW

File: rollo/algorithm.py [hunk 1]
```diff
@@ -182,87 +144,71 @@ def initialize_pop(self, pop):
...
         Parameters
         ----------
         pop : list
             list of deap.creator.Ind for previous generation
         gen: int
             generation number
-
         Returns
         -------
         list
```
Comment (smpark7) [inline rollo/algorithm.py:155]: Should this be `pop : list` instead?

File: rollo/toolbox_generator.py [hunk 2]
```diff
@@ -33,11 +38,11 @@ def setup(self, evaluator_fn, input_algorithm, input_ctrl_vars, control_dict):
         """
 
         weight_list = []
-        for obj in input_algorithm["objective"]:
+        for obj, wt in zip(input_algorithm["objective"], input_algorithm["weight"]):
             if obj == "min":
-                weight_list.append(-1.0)
+                weight_list.append(-wt)
```
Comment (samgdotson) [inline rollo/toolbox_generator.py:43]: A good clean change, does this fix a bug?

File: rollo/algorithm.py [hunk 3]
```diff
@@ -126,30 +98,24 @@ def generate(self):
...
         """Initialize population for genetic algorithm
-
         Parameters
         ----------
         pop : list
             list of deap.creator.Ind for previous generation
-
         Returns
         -------
         list
```
Comment (smpark7) [inline rollo/algorithm.py:117]: Should this be `pop : list`?

**Review by samgdotson — COMMENTED** [review]
I don't have any serious comments on this, besides pointing out that there are some open issues that this PR doesn't seem to address.

**Comment by samgdotson** [comment]
Oh, also the CI fails.

File: rollo/algorithm.py [hunk 4]
```diff
@@ -182,87 +144,71 @@ def initialize_pop(self, pop):
...
-        pop = self.apply_mating_operator(pop)
-        pop = self.apply_mutation_operator(pop)
+        offspring = self.apply_mating_operator(pop)
+        offspring = self.apply_mutation_operator(offspring)
         # define pop's gen, ind num
-        for i, ind in enumerate(pop):
+        for i, ind in enumerate(offspring):
             ind.gen = gen
             ind.num = i
         # evaluate fitness of newly created pop for inds with invalid fitness
```
Comment (smpark7) [inline rollo/algorithm.py:165]: Update doc comment here.

File: rollo/algorithm.py [hunk 5]
```diff
@@ -182,87 +144,71 @@ def initialize_pop(self, pop):
...
+                    self.toolbox.evaluate,
+                    list(invalids)))
         # assign fitness values to individuals
         for ind, fitness in zip(invalids, fitnesses):
             fitness_vals = []
             for i in range(self.toolbox.objs):
                 fitness_vals.append(fitness[i])
             ind.fitness.values = tuple(fitness_vals)
             ind.output = fitness
+        pop = self.apply_selection_operator(pop+offspring)
```
Comment (smpark7) [inline rollo/algorithm.py:182]: Add comment here if necessary

**Comment by samgdotson** [comment]
I tried creating a commit to your github action -- I think all you need to do is append `unit_tests/` to each of your pytest calls. My attempt at a PR to your fork got rejected for some reason.

File: rollo/backend.py [hunk 6]
```diff
@@ -191,9 +191,9 @@ def update_backend(self, pop, gen, invalid_ind, rndstate):
                         self.input_file["evaluators"][solver]["output_script"], "r"
                     ) as file:
                         evaluator_files[solver + "_output"] = file.read()
-                except:
+                except BaseException:
                     pass
```
Comment (smpark7) [inline rollo/backend.py:195]: If this is intended behavior, you may wanna add a print statement here stating that it passed and why. If you're gonna implement smth here in future, then possibly add a `# TODO: ...` comment.

File: rollo/executor.py [hunk 7]
```diff
@@ -185,16 +186,23 @@ def load_evaluator(self, control_dict, output_dict, input_dict):
             solver_dict = input_evaluators[solver]
             try:
                 output_script = solver_dict["output_script"]
-            except:
+            except BaseException:
```
Comment (smpark7) [inline rollo/executor.py:189]: It may be helpful to add a print statement here too, to indicate that the exception occurred

**Review by smpark7 — CHANGES_REQUESTED** [review]
This is a basic review for code clealiness. Good job! Your code looks mostly clean. In addition to the comments, I would suggest adding print statements in input_validation to inform the user of whichever parameters were initialized to default values (if you haven't already done this).

**Pull request closed without merging**